### PR TITLE
Adding support for linux/ppc64le in github action for kserve-controller

### LINKED ISSUE #2574: add support for linux-ppc64le arch in CI  for kserve-controller
Hi there,
The changes for llinux-ppc64le are already in kserve-controller Dockerfile. 
please review and merge the following PR to release docker image for linux-ppc64le using multi-arch docker image.
[PR link](https://github.com/kserve/kserve/pull/2550#issuecomment-1328040115)

### PR DESCRIPTION
**Pull request by adilhusain-s**
**What this PR does / why we need it**:
This pr enables github action to generate the multi-arch docker image for kserve-controller.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2574 

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by yuzisun** [comment]
@adilhusain-s Thanks for the PR, can you help add more description for what's the changes about and if his changes the image we publish for KServe ?

**Comment by adilhusain-s** [comment]
@yuzisun 
no, it doesn't change linux-amd64 docker image in anyway. We're building multi-arch docker images here.
Please read official  [docs](https://docs.docker.com/build/building/multi-platform/).

**Comment by yuzisun** [comment]
> @yuzisun
> no, it doesn't change linux-amd64 docker image in anyway. We're building multi-arch docker images here.
> Please read official [docs](https://docs.docker.com/build/building/multi-platform/).

Thanks for the docs! Can you help create a corresponding kserve issue and link to this PR?

**Comment by adilhusain-s** [comment]
@yuzisun 
As requested, created the issue.
[issue link](https://github.com/kserve/kserve/issues/2574)

**Comment by pranavpandit1** [comment]
@yuzisun - Any Updates on this request, I'd be interested in these changes.

**Comment by yuzisun** [comment]
> @yuzisun - Any Updates on this request, I'd be interested in these changes.

I think we will also want to update for `router` and `qpext` images 
https://github.com/kserve/kserve/blob/master/.github/workflows/router-docker-publish.yml
https://github.com/kserve/kserve/blob/master/.github/workflows/qpext-docker-publish.yml

**Comment by adilhusain-s** [comment]
@yuzisun 

I'll raise the PRS for router and qpext  component. could you please merge this one?

**Comment by adilhusain-s** [comment]
@yuzisun 
as you requested, the PRS have been raised for qpext and router.
[qpext PR](https://github.com/kserve/kserve/pull/2604)
[router pr](https://github.com/kserve/kserve/pull/2605)

File: .github/workflows/kserve-controller-docker-publish.yml [hunk 1]
```diff
@@ -46,14 +46,21 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Build image
-        run: |
-          docker build . --file Dockerfile --tag $IMAGE_NAME
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: ppc64le
```
Comment (ddelange) [inline .github/workflows/kserve-controller-docker-publish.yml]:
```suggestion
```

File: .github/workflows/kserve-controller-docker-publish.yml [hunk 2]
```diff
@@ -69,9 +76,14 @@ jobs:
...
-          docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
-          docker push $IMAGE_ID:$VERSION
-
+          echo VERSION=$VERSION >> $GITHUB_ENV
+          echo IMAGE_ID=$IMAGE_ID >> $GITHUB_ENV
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          platforms: linux/amd64,linux/ppc64le
```
Comment (ddelange) [inline .github/workflows/kserve-controller-docker-publish.yml]:
```suggestion
          platforms: linux/amd64,linux/arm/v7,linux/arm64/v8,linux/ppc64le,linux/s390x
```

could you please include these architecutres? for feature parity with knative serving:

```
$ docker run --rm mplatform/mquery gcr.io/knative-releases/knative.dev/serving/cmd/activator@sha256:9cb7c468f0975079c7b86d1b2b6b61dd9b5e64d38cd92ccb5239348ce5bd9dee
Image: gcr.io/knative-releases/knative.dev/serving/cmd/activator@sha256:9cb7c468f0975079c7b86d1b2b6b61dd9b5e64d38cd92ccb5239348ce5bd9dee (digest: sha256:9cb7c468f0975079c7b86d1b2b6b61dd9b5e64d38cd92ccb5239348ce5bd9dee)
 * Manifest List: Yes (Image type: application/vnd.docker.distribution.manifest.list.v2+json)
 * Supported platforms:
   - linux/amd64
   - linux/arm
   - linux/arm64
   - linux/ppc64le
   - linux/s390x
```

**Comment by adilhusain-s** [comment]
@yuzisun @ddelange 

I've just accepted the suggested changes and force pushed. could you please review and merge?

File: .github/workflows/kserve-controller-docker-publish.yml [hunk 3]
```diff
@@ -69,9 +74,14 @@ jobs:
...
           [ "$VERSION" == "master" ] && VERSION=latest
 
-          echo IMAGE_ID=$IMAGE_ID
-          echo VERSION=$VERSION
-
-          docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
-          docker push $IMAGE_ID:$VERSION
-
+          echo VERSION=$VERSION >> $GITHUB_ENV
+          echo IMAGE_ID=$IMAGE_ID >> $GITHUB_ENV
```
Comment (ddelange) [inline .github/workflows/kserve-controller-docker-publish.yml:78]: a nit to be safe:
```suggestion
          echo "VERSION=$VERSION" >> $GITHUB_ENV
          echo "IMAGE_ID=$IMAGE_ID" >> $GITHUB_ENV
```

**Review by ddelange — COMMENTED** [review]
there's still the DCO check failing, for the rest LGTM 👍

**Comment by yuzisun** [comment]
@adilhusain-s can you help sign off the commit ?

**Pull request merged**